src/depackers/vorbis.c

```diff
@@ -2299,14 +2296,7 @@ static int decode_residue(vorb *f, float *residue_buffers[], int ch, int n, int
                      float *target = residue_buffers[j];
                      int offset = r->begin + pcount * r->part_size;
                      int n = r->part_size;
-                     Codebook *book;
-
-                     /* Sanity check */
-                     if (offset + n >= f->blocksize_1) {
-                        return FALSE;
-                     }
-
-                     book = f->codebooks + b;
+                     Codebook *book = f->codebooks + b;
```

Comment:
I alluded to this check not helping in #466, but here's the reason why:

1) `offset` = `r->begin + pcount * r->part_size`. (line 2300)
2) 0 <= `pcount` < `part_read`. (line 2289)
3) `part_read` = `n_read / r->part_size`. (line 2153)
4) `n_read` = `limit_r_end - limit_r_begin`, both clamped to range [0, `blocksize`]. (lines 2149-2152)
5) `n` (in this particular block) = `r->part_size`. (line 2301)
6) `(part_read - 1)` (maximum value of `pcount`) ` * r->part_size + r->part_size` (value of `n`) == `part_read * r->part_size` <= `n_read` <= `blocksize`, so this check isn't necessary, and worst case may be able to refuse valid offset+length pairs due to off-by-one in the check equality.